### PR TITLE
Added dryRun flag for printing docker command for demo and sandbox

### PR DESCRIPTION
**Pull request by pmahindrakar-oss**
Signed-off-by: pmahindrakar-oss <prafulla.mahindrakar@gmail.com>

# TL;DR

Command : 

```
flytectl demo start --dryRun  
```

O/P 

```
❇️ Run the following command to remove the existing sandbox
	docker container rm flyte-sandbox --force
🐋 Going to use Flyte v1.2.0 release with image cr.flyte.org/flyteorg/flyte-sandbox-lite:sha-4f73dc6994dfeafb9eecd9b17d16d7f9275b577a 
❇️ Run the following command to pull the sandbox image from registry.
	docker pull  cr.flyte.org/flyteorg/flyte-sandbox-lite:sha-4f73dc6994dfeafb9eecd9b17d16d7f9275b577a 
❇️ Run the following command to create new sandbox container
	docker create --privileged -p 0.0.0.0:30081:30081 -p 0.0.0.0:30082:30082 -p 0.0.0.0:30084:30084 -p 0.0.0.0:30086:30086 -p 0.0.0.0:30088:30088 -p 0.0.0.0:30089:30089 -p 0.0.0.0:30090:30090 -p 0.0.0.0:30080:30080 --env SANDBOX=1 --env KUBERNETES_API_PORT=30086 --env FLYTE_HOST=localhost:30081 --env FLYTE_AWS_ENDPOINT=http://localhost:30084 --mount type=bind,source=/Users/praful/.flyte,target=/etc/rancher/ --name flyte-sandbox cr.flyte.org/flyteorg/flyte-sandbox-lite:sha-4f73dc6994dfeafb9eecd9b17d16d7f9275b577a
❇️ Run the following command to start the sandbox container
	docker start flyte-sandbox
✳️ Run the following command to check the logs and monitor the sandbox container and make sure there are no error during startup and then visit flyteconsole
	docker logs -f flyte-sandbox
👨‍💻 Flyte would be ready after this! Flyte UI would be available at http://localhost:30080/console 🚀 🚀 🎉 
❇️ Run the following command to export sandbox environment variables for accessing flytectl
	Add FLYTECTL_CONFIG to your environment variable 
❇️ Run the following command to export kubeconfig variables for accessing flyte pods locally
	export KUBECONFIG=$KUBECONFIG:/Users/praful/.kube/config:/Users/praful/.flyte/k3s/k3s.yaml 
```
## Type
- [ ] Bug Fix
- [X] Feature
- [ ] Plugin

## Are all requirements met?

- [ ] Code completed
- [ ] Smoke tested
- [ ] Unit tests added
- [ ] Code documentation added
- [ ] Any pending items have an associated Issue

## Complete description
_How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/3041

## Follow-up issue
_NA_

### CODE REVIEW

**Review by wild-endeavor — APPROVED** [review]
I think --dry-run would be a better name for the switch but up to you if you want to change it.  i'm a bit worried about this pr though, we can sync offline.

**Review by wild-endeavor — APPROVED** [review]

**Review by EngHabu — COMMENTED** [review]
Is there no way to use the existing client to generate the command line? all this string concat seem error prune..

File: cmd/config/subcommand/sandbox/sandbox_config.go [hunk 1]
```diff
@@ -29,7 +29,8 @@ type Config struct {
 	ImagePullOptions docker.ImagePullOptions `json:"imagePullOptions" pflag:",Optional. Defines image pull options (e.g. auth)"`
 
 	// It's used for development. Users are able to start flyte locally via single binary and save the data to the minio or postgres in the sandbox.
-	Dev bool `json:"dev" pflag:",Optional. Only start minio and postgres in the sandbox."`
+	Dev          bool `json:"dev" pflag:",Optional. Only start minio and postgres in the sandbox."`
+	PrintCommand bool `json:"printCommand" pflag:",Optional. Only print the command to start the docker container."`
```
[the remark this answers is not in the document]
Comment (pmahindrakar-oss) [reply cmd/config/subcommand/sandbox/sandbox_config.go:33]: docker has the following interface for interacting with API 
https://github.com/moby/moby/blob/master/client/interface.go

But has only one implementation which is there concrete implementation which calls into the docker service  eg : https://github.com/moby/moby/blob/master/client/container_create.go

We can add implementations for these but that would still mean doing some sort of string concatenation unless the service support converting these to docker command which they dont atm

**Comment by pmahindrakar-oss** [comment]
@EngHabu  @wild-endeavor 
docker has the following interface for interacting with API
https://github.com/moby/moby/blob/master/client/interface.go

But has only one implementation which is there concrete implementation which calls into the docker service eg : https://github.com/moby/moby/blob/master/client/container_create.go

We can add implementations for these but that would still mean doing some sort of string concatenation unless the service support converting these to docker command which they dont atm

I can change from printCommand to dryRun

File: cmd/config/subcommand/sandbox/sandbox_config.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (pmahindrakar-oss) [reply cmd/config/subcommand/sandbox/sandbox_config.go]: done

**Review by wild-endeavor — DISMISSED** [review]

**Comment by wild-endeavor** [comment]
+1 to @EngHabu... but we already committed to doing this so it's okay.  but yeah, this feels like the maintenance burden it adds is not worth it.  I'd rather just print out the arguments to each cli.Docker function call with a series of log statements.

**Comment by pmahindrakar-oss** [comment]
@wild-endeavor we dont have to commit to this approach if we find a better alternative.
Dumping the function call args would still mean the user has to map those to corresponding docker command args .
If we are ok with that they that makes this PR much simpler . cc : @kumare3

Ideally a common interface across docker cli and API would have solved this but thats not the case. 

I understand the maintenance of this path but would have been great if docker API's directly supported some sort of API to commandline dry run option which they don't.
I am holding of the merge until we come to a conclusion on this.

**Pull request merged**